### ISSUE TITLE
INVESTMENT FLOW - Invest entire balance if user types visible balance

### ISSUE THREAD

**Issue by Desperado** [Neufund/platform-frontend#2623]
**Background**
We round down wallet balances to two decimals. When user types such balance in the redeem modal, we currently invest amount typed so the remainder will still be in the wallet. Example
visible balance: 1202312.84 nEUR (also rounded incorrectly as in #2557)
entire balance: 1202312.83426 nEUR
remainder 0.00426 nEUR

**Technical notes**
This is 100% similar to the changes in REDEEM Flow #2601

**Acceptance criteria**
After typing visible balance, entire balance is withdrawn and new nEUR balance is exactly 0
Of course, "Invest Your Entire Wallet Balance" should work fine as well.

**Testing notes**
Best way to test is of course to use etherscan and just see the account. Alternatively you can see nEUR token balance in the Metamask when you add it as a token. Please ask any developer to find nEUR address for you. it will be in transaction posted to pending pool.
Also testable by checking the redeemed amount in the backoffice (it should be higher/full despite the rounding-down visible for the user)

![Selection_046.png](https://images.zenhubusercontent.com/5c3ddac32c893f1dad8d7a9a/4f81f983-8325-4341-ba2e-af4401a25004)

**Comment by glupiawymowka** [comment]
this is almost correct, but the ETH total with GAS is still rounded up : 
 
balance in wallet: 
![image](https://user-images.githubusercontent.com/18315589/55341979-f77dd000-54a7-11e9-988a-698accf8fb38.png)

![image](https://user-images.githubusercontent.com/18315589/55341970-ee8cfe80-54a7-11e9-9f6a-77706ff9f914.png)

- [ ] I've used entire balance for investment,as you can see in the total above,  but I still have 0.0018 ETH left after this investment
![image](https://user-images.githubusercontent.com/18315589/55342104-488dc400-54a8-11e9-84ec-adb719db152e.png)

**Comment by drakkein** [comment]
I did some tests for this issue when some amount is still left after investing whole balance. It's working as it's designed. CC @rudolfix @Moejoe90 @desfero @AlexeyTuboltsev 

Let me quickly explain. 
I've created transaction with whole balance (initially available was `2707.013139 ETH`). Input value was set to `2707.011411 ETH` which is **available balance - estimated gas price** (calculated to `0.001728 ETH` displayed as `0.0018 ETH`). After confirmation of transaction we can see that we have some ETH left. 
![image](https://user-images.githubusercontent.com/10601291/55433900-a7316b80-5596-11e9-8a0f-c76d7335c71e.png)

In transaction details we can see that total ETH used for it is only `2707.012172 ETH`

![image](https://user-images.githubusercontent.com/10601291/55434009-dba52780-5596-11e9-940e-695f8354f268.png)

If we calculate it `2707.012172 ETH` - `2707.011411 ETH` = `0.000761 ETH` <- real gas cost
So `0.001728 ETH` - `0.000761 ETH` gives us amount that has been left which is `0.000967 ETH`.

For now I have no idea how we can use whole amount in case of ETH wallet.
